### ISSUE TITLE
Adding Cypress to the project

### ISSUE THREAD

**Issue by iusehooks** [iusehooks/usetheform#24]
We would like to add Cypress https://www.cypress.io/ to the project and write some automation tests.

**Comment by Loriage** [comment]
Hello there! 
I'm new to open-source contribution and I'm interested by your issue.

Can I take a look at it?
Thanks in advance

**Comment by iusehooks** [comment]
Hi @Loriage , 

You are welcome. You can take a look into it.

Thanks.

**Comment by iusehooks** [comment]
Hi @Loriage , 

It should run using cypress cli.

we should create a new scripts:

```javascript
 "scripts": {
     "cypress": "cypress run"
 }
```
no need for running in a website. Hope it helped.

**Comment by Loriage** [comment]
Hi,

Of course for the new script! My question is: Should I run the test on a demo website/page. Usetheform is a node module if I have well understand, not a react application itself. So if  I don't misunderstand, cypress is a webpage tester, no?

**Comment by iusehooks** [comment]
Hi @Loriage,

I got what you meant now. Tests should run locally.

Thanks.

**Comment by Loriage** [comment]
Hello @iusehooks,

I'm facing an issue when I'm trying to run a react app with the content of the repo as module.
Here the error:

src/usetheform/hooks/useField.js
  Line 465:27:  React Hook "useRef" is called in function "getInitialValue" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter  react-hooks/rules-of-hooks
  Line 466:29:  React Hook "useRef" is called in function "getInitialValue" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter  react-hooks/rules-of-hooks
  
Any ideas?
 
Thanks.

**Comment by iusehooks** [comment]
Hi @Loriage,

may you copy and paste some code example?

Thanks.

**Comment by Loriage** [comment]
@iusehooks Of course!

./testers/App.js:
`import React from "react";
import { Form, Input, useValidation } from "./usetheform";

const preventNegativeNumber = (next, prev) => (next <= 0 ? 0 : next);
const required = (value) =>
    value && value.trim() !== "" ? undefined : "Required";

export default function App() {
  const onChange = (formState) => console.log("ON_CHANGE : ", formState);
  const onSubmit = (formState) => console.log("ON_SUBMIT : ", formState);

  const [status, validation] = useValidation([required]);

  return (
    <Form onSubmit={onSubmit} onChange={onChange}>
      <Input name="firstname" type="text" touched {...validation} />
      {status.error && <span>{status.error}</span>}
      <Input name="lastname" type="text" />
      <Input name="age" type="number" value={18} reducers={preventNegativeNumber} />
      <button type="submit">Submit</button>
    </Form>
  );
}`

Here my installation script:
 `
#!/bin/env sh
cp -r src/ testers/src/usetheform    \
&& cd testers                                 \
&& npm start`

So basically, I copy-paste the sourcefiles of usetheform to my react app, loading it as module and run the app.

Thanks.

**Comment by iusehooks** [comment]
Would you also please take a screen shoot of the project structure folders in your IDE?

**Comment by Loriage** [comment]
Here my installation script:
 `
#!/bin/env sh
cp -r src/ testers/src/usetheform    \
&& cd testers                                 \
&& npm start`

So basically, I copy-paste the sourcefiles of usetheform to my react app, loading it as module and run the app.

Thanks.

**Comment by Loriage** [comment]
Yes of course
![image](https://user-images.githubusercontent.com/32834445/111832583-491c0100-88f1-11eb-9f1f-0e8b6370f965.png)
![image](https://user-images.githubusercontent.com/32834445/111832606-546f2c80-88f1-11eb-817e-6e708a753576.png)

**Comment by Loriage** [comment]
testers/usrc/usetheform/ is a copy of src/

**Comment by iusehooks** [comment]
Would you please try to run in your local npm run build command and than copy and paste the /build/index.js file into src folder and rename it as usetheform.js, so you can remove the usetheform folder.

**Comment by Loriage** [comment]
Seems working a very short time but display this error 0.5second after:
![image](https://user-images.githubusercontent.com/32834445/111833468-81700f00-88f2-11eb-981b-2a628f68f503.png)

**Comment by iusehooks** [comment]
would you please start a clean project downloading it from: https://codesandbox.io/s/still-hooks-tzodd ?

**Comment by Loriage** [comment]
Uhm, facing prettier error now:
![image](https://user-images.githubusercontent.com/32834445/111834469-ed9f4280-88f3-11eb-9ac5-257e95dce8e6.png)

**Comment by Loriage** [comment]
The issue is how to launch a testable react app inside the usetheform directory without facing conflict between the repository and the test app.

**Comment by iusehooks** [comment]
I think you do not need to use creare-react-app. 

I would suggest you to look into the folder named examples and take inspiration of that.  

You can excute the command "npm run dev" and than you may create a new .html file inside the folder named examples, then you are ok to start writing tests in cypress. A web server runs at http://localhost:3000/{yourfile}.html

hope it helped.

**Comment by Loriage** [comment]
Working, thanks. I'll continue my tests!

**Comment by iusehooks** [comment]
Hi @Loriage, 

which approach did you go for? 

If you like the library may you please give a github star to it? It helps the project to grow.

Thanks.

**Comment by Loriage** [comment]
Hi @iusehooks,
I'll made my tests on the dev web server

**Comment by Loriage** [comment]
Hello @iusehooks !

First PR here: https://github.com/iusehooks/usetheform/pull/33

Waiting for your feedback!

**Comment by Khemendra-Bhardwaj** [comment]
Hey there, is it open so I can start working on this issue.

**Comment by Ansh1372** [comment]
Hey I'm new to open source contribution can you asign this challenge that can I enhance my skill...

**Comment by SavvasMichaelTruong** [comment]
Hi May I work on this issue to add Cypress?

**Comment by m-truong** [comment]
Hello @iusehooks ,

I found this from GoodFirstIssues, and this project seems very interesting to me, is this still open?

I’m actually just working on adding Cypress to my company’s NestJS backend server with GraphQL. 

I’m eager to help contribute and add Cypress to this project ᕦ(ò_óˇ)ᕤ . Please assign this issue to me? Thanks!

@iusehooks

**Comment by MayankMaloo192** [comment]
Hey @iusehooks  is this issue still open? I want to contribute to this issue .
 Can you please assign it to me.

**Comment by AliSinaYOusofi** [comment]
Hi
Is the issue still open ?

**Comment by MuhammadAwaisGondal** [comment]
Hey There, I am new to open source and I found this on GoodFirstIssues so can I give it a try? 
Thank you

**Comment by AshwinNema** [comment]
Hi @antoniopangallo  ,
I am new to open source. I am interested in solving this issue. Can it be assigned to me?

**Comment by AshwinNema** [comment]
Hi @iusehooks,

Can I take up this issue?

**Comment by AshwinNema** [comment]
Hi @iusehooks,

I have added remaining tests in my PR. Please let me know if anything has to be changed.

**Comment by m-truong** [comment]
Hey ~ i've been busy with work,

But can I get this reassigned to me to complete the cypress suite?

**Comment by SiddheshTambe45** [comment]
Hi! I'd like to work on this issue. Could I be assigned to it?
